jsr311.go

```diff
@@ -151,20 +149,16 @@ func (r RouterJSR311) detectRoute(routes []Route, httpRequest *http.Request) (*R
 		for _, candidate := range previous {
 			available = append(available, candidate.Produces...)
 		}
-		// if POST,PUT,PATCH without body
-		method, length := httpRequest.Method, httpRequest.Header.Get("Content-Length")
-		if (method == http.MethodPost ||
-			method == http.MethodPut ||
-			method == http.MethodPatch) && (length == "" || length == "0") {
+		method := httpRequest.Method
+		if httpRequest.ContentLength > 0 || method == http.MethodPost || method == http.MethodPut || method == http.MethodPatch {
 			return nil, NewError(
 				http.StatusUnsupportedMediaType,
 				fmt.Sprintf("415: Unsupported Media Type\n\nAvailable representations: %s", strings.Join(available, ", ")),
```

Comment:
The `if` block above (where `candidates` is filtered to 0-length based on the content-type header) definitely seems like a clearer place to return 415 errors

Since this block is about filtering based on the `Accept` header, which should return 406 errors, I liked how the previous PR iteration moved the 415 errors out of this block.